src/base/utils/string.h

```diff
@@ -36,19 +36,45 @@
 #include <QString>
 #include <Qt>
 #include <QtContainerFwd>
+#include <QtGlobal>
+
+#ifndef Q_OS_WIN
+#define QBT_USE_QCOLLATOR
+#include <QCollator>
+#endif
 
 class QStringRef;
 
 namespace Utils::String
 {
-    QString fromDouble(double n, int precision);
+#ifndef QBT_USE_QCOLLATOR
```

Comment:
There are too many conditionals. IMO, there's nothing wrong if this function exists unconditionally.

Reply:
If this function exists unconditionally, I think there are too many options exposed. 
And this function only exists when QCollator is not suitable, thus it should be an implementation detail.

Reply:
>And this function only exists when QCollator is not suitable, thus it should be an implementation detail.

Then why not move it in NaturalCompare class?

Reply:
>Then why not move it in NaturalCompare class?

Then it would need to move the code into header.